internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/install.go

```diff
@@ -62,26 +66,63 @@ func (c *Config) Install(opts options.InstallOptions) (*options.InstallResult, e
 	return installResult, err
 }
 
-//nolint:gocyclo
+//nolint:gocyclo,cyclop
 func (c *Config) install(opts options.InstallOptions) (*options.InstallResult, error) {
 	if err := c.flip(); err != nil {
 		return nil, err
 	}
 
-	if err := utils.CopyFiles(
-		opts.Printf,
-		utils.SourceDestination(
-			opts.BootAssets.KernelPath,
-			filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.KernelAsset),
-		),
-		utils.SourceDestination(
-			opts.BootAssets.InitramfsPath,
-			filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.InitramfsAsset),
-		),
-	); err != nil {
-		return nil, err
+	if quirks.New(opts.Version).TalosVersionSDBootOnly() {
+		assetInfo, err := uki.Extract(opts.BootAssets.UKIPath)
+		if err != nil {
+			return nil, err
+		}
+
+		defer assetInfo.Close() //nolint:errcheck
+
+		if err := os.MkdirAll(filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default)), 0o755); err != nil {
+			return nil, err
+		}
+
+		kernel, err := os.Create(filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.KernelAsset))
+		if err != nil {
+			return nil, err
+		}
+
+		defer kernel.Close() //nolint:errcheck
+
+		initrd, err := os.Create(filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.InitramfsAsset))
+		if err != nil {
+			return nil, err
+		}
+
+		defer initrd.Close() //nolint:errcheck
+
+		if _, err := io.Copy(kernel, assetInfo.Kernel); err != nil {
+			return nil, err
+		}
+
+		if _, err := io.Copy(initrd, assetInfo.Initrd); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := utils.CopyFiles(
+			opts.Printf,
+			utils.SourceDestination(
+				opts.BootAssets.KernelPath,
+				filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.KernelAsset),
+			),
+			utils.SourceDestination(
+				opts.BootAssets.InitramfsPath,
+				filepath.Join(opts.MountPrefix, constants.BootMountPoint, string(c.Default), constants.InitramfsAsset),
+			),
+		); err != nil {
+			return nil, err
+		}
 	}
 
+	// TODO(frezbo): should we merge the cmdline from the UKI file with the one from the options?
```

Comment:
I think we can ignore the uki cmdline, since this will be an upgrade path in most cases unless booting bios

Reply:
yes, in GRUB case it's explicit I think